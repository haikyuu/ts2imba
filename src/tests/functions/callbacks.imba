import {build} from "../../index"
import * as imbac from 'imba/compiler'

let tsx-code = '''
useEffect(() => {
	ref.current = value
}, [value])
'''

test 'callbacks as first params are handled well' do
	const result = await build tsx-code
	expect(result.code).toMatchSnapshot!
	const out = imbac.compile(result.code, sourceId: 'sth')

let tsx-code1 = '''
useEffect([value], () => {
	ref.current = value
})
'''

test.only 'callbacks as last params are handled well' do
	const result = await build tsx-code1
	expect(result.code).toMatchSnapshot!
	const out = imbac.compile(result.code, sourceId: 'sth')

