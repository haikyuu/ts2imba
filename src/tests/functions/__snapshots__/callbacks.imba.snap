// Vitest Snapshot v1

exports[`callbacks as last params are handled well 1`] = `
"useEffect([value]) do
	ref.current = value
"
`;
